tensorboard/plugins/graph/graph_util.py

```
@@ -0,0 +1,214 @@
+  Functional:
+    config
+      name: Name of the model. If not specified, it is 'model' with
+            an optional suffix if there are more than one instance.
+      input_layers: Keras.layers.Inputs in the model.
+      output_layers: Layer names that are outputs of the model.
+      layers: list of layer configurations.
+        layer: [*]
+          inbound_nodes: inputs to this layer.
+
+  Sequential:
+    config
+      name: Name of the model. If not specified, it is 'sequential' with
+            an optional suffix if there are more than one instance.
+      layers: list of layer configurations.
+        layer: [*]
+
+  [*]: Note that a model can be a layer.
+  """
+  layer_config = model_layer.get('config')
+  if not layer_config.get('layers'):
+    raise ValueError('layer is not a model.')
+
+  node_name = _scoped_name(name_scope, layer_config.get('name'))
+  input_layers = layer_config.get('input_layers')
+  output_layers = layer_config.get('output_layers')
+  inbound_nodes = model_layer.get('inbound_nodes')
+
+  is_functional_model = bool(input_layers and output_layers)
+  # In case of [1] and the parent model is functional, current layer
+  # will have the 'inbound_nodes' property.
+  is_parent_functional_model = bool(inbound_nodes)
+
+  if is_parent_functional_model and is_functional_model:
+    for (input_layer, inbound_node) in zip(input_layers, inbound_nodes):
+      input_layer_name = _scoped_name(node_name, input_layer)
+      inbound_node_name = _scoped_name(name_scope, inbound_node[0])
+      input_to_in_layer[input_layer_name] = inbound_node_name
+  elif is_parent_functional_model and not is_functional_model:
+    # Sequential model can take only one input. Make sure inbound to the
+    # model is linked to the first layer in the Sequential model.
+    prev_node_name = _scoped_name(name_scope, inbound_nodes[0][0])
+  elif not is_parent_functional_model and prev_node_name and is_functional_model:
+    assert len(input_layers) == 1, (
```

Do you want to keep this assertion here, or raise an Exception? Is this truly completely unexpected? Will the assertion completely break the graph plugin?

---

It is truly unexpected and imo, it is incorrect to show a graphdef. I was literally thinking of breaking the application in that case.